geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/membership/MembershipOnlyTest.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.geode.distributed.internal.membership.gms.api.MembershipConfig;
+import org.apache.geode.distributed.internal.tcpserver.TcpClient;
+import org.apache.geode.distributed.internal.tcpserver.TcpSocketCreator;
+import org.apache.geode.internal.InternalDataSerializer;
+import org.apache.geode.internal.admin.SSLConfig;
+import org.apache.geode.internal.net.SocketCreator;
+import org.apache.geode.internal.serialization.DSFIDSerializer;
+
+public class MembershipOnlyTest {
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private InternalLocator locator;
+  private InetAddress localHost;
+
+  @Before
+  public void before() throws IOException {
+    localHost = InetAddress.getLocalHost();
+
+    // TODO - using a geode-core locator
+    locator = InternalLocator.startLocator(0, new File(""), null, null, localHost, false,
+        new Properties(), null, temporaryFolder.getRoot().toPath());
+  }
+
+  @After
+  public void after() {
+    locator.stop();
+  }
+
+  @Test
+  public void locatorStarts() {
+    assertThat(locator.getPort()).isGreaterThan(0);
+  }
+
+  @Test
+  public void memberCanConnectToLocator() throws MemberStartupException {
+
+    final MembershipConfig config = new MembershipConfig() {
+      public String getLocators() {
+        return localHost.getHostName() + '[' + locator.getPort() + ']';
+      }
+    };
+
+    // TODO - using geode-core InternalDistributedMember
+    MemberIdentifierFactoryImpl memberIdFactory = new MemberIdentifierFactoryImpl();
+
+    // TODO - using geode-core serializer. This is needed to have be able to
+    // read InternalDistributedMember.
+    DSFIDSerializer dsfidSerializer = InternalDataSerializer.getDSFIDSerializer();
+
+    // TODO - using geode-core socket creator
+    final TcpSocketCreator socketCreator =
+        asTcpSocketCreator(new SocketCreator(new SSLConfig.Builder().build()));
+
+    TcpClient client = new TcpClient(socketCreator, dsfidSerializer.getObjectSerializer(),
+        dsfidSerializer.getObjectDeserializer());
+
+    LifecycleListener<InternalDistributedMember> lifeCycleListener = mock(LifecycleListener.class);
+
+    final Membership<InternalDistributedMember> membership =
+        MembershipBuilder.<InternalDistributedMember>newMembershipBuilder()
+            .setConfig(config)
+            .setSerializer(dsfidSerializer)
+            .setLifecycleListener(lifeCycleListener)
+            .setMemberIDFactory(memberIdFactory)
+            .setLocatorClient(client)
+            .setSocketCreator(socketCreator)
+            .create();
+
+
+    // TODO - the membership *must* be installed in the locator at this special
+    // point during membership startup for the start to succeed
+    doAnswer(invocation -> {
+      locator.getLocatorHandler().setMembership(membership);
+      return null;
+    }).when(lifeCycleListener).started();
+
+
+    membership.start();
+    assertThat(membership.getView().getMembers()).hasSize(1);
```

Comment:
this fails the ArchUnit test as written, need a resolution or a temporary exception in the test...